src/algorithms/libs/beidou_b1c_signal_processing.cc

```diff
@@ -0,0 +1,859 @@
+
+            //--- Make index array to read B1C pilot code values -------------------------
+            _codeValueIndex = ceil((_ts * (static_cast<float>(ii) + 1)) / _tc) - 1;
+
+            //--- Make the digitized version of the B1C code -----------------------
+            if (ii == _samplesPerCode - 1)
+                {
+                    //--- Correct the last index (due to number rounding issues) -----------
+                    _dest[ii] = std::complex<float>(1.0 - 2.0 * _code_data[_codeLength - 1], 1.0 - 2.0 * _code_pilot[_codeLength - 1]);
+                }
+            else
+                {
+                    _dest[ii] = std::complex<float>(1.0 - 2.0 * _code_data[_codeValueIndex], 1.0 - 2.0 * _code_pilot[_codeValueIndex]);  //repeat the chip -> upsample
+                }
+        }
+}
+
+
+//================================================BDS_B1C_BOC_Generation===================================================
+
+
+void beidou_b1c_data_sinboc_11_gen_int(gsl::span<int> _dest, gsl::span<const int> _prn)
+{
+    const uint32_t _length_in = BEIDOU_B1Cd_CODE_LENGTH_CHIPS;
+    auto _period = static_cast<uint32_t>(_dest.size() / _length_in);
+    for (uint32_t i = 0; i < _length_in; i++)
+        {
+            for (uint32_t j = 0; j < (_period / 2); j++)
+                {
+                    _dest[i * _period + j] = _prn[i];
+                }
+            for (uint32_t j = (_period / 2); j < _period; j++)
+                {
+                    _dest[i * _period + j] = -_prn[i];
+                }
+        }
+}
+
+
+void beidou_b1c_pilot_sinboc_11_gen_int(gsl::span<int> _dest, gsl::span<const int> _prn)
+{
+    const uint32_t _length_in = BEIDOU_B1Cp_CODE_LENGTH_CHIPS;
+    auto _period = static_cast<uint32_t>(_dest.size() / _length_in);
+    for (uint32_t i = 0; i < _length_in; i++)
+        {
+            for (uint32_t j = 0; j < (_period / 2); j++)
+                {
+                    _dest[i * _period + j] = _prn[i];
+                }
+            for (uint32_t j = (_period / 2); j < _period; j++)
+                {
+                    _dest[i * _period + j] = -_prn[i];
+                }
+        }
+}
+
+
+void beidou_b1c_pilot_sinboc_61_gen_int(gsl::span<int> _dest, gsl::span<const int> _prn)
+{
+    const uint32_t _length_in = BEIDOU_B1Cp_CODE_LENGTH_CHIPS;
+    auto _period = static_cast<uint32_t>(_dest.size() / _length_in);
+
+    for (uint32_t i = 0; i < _length_in; i++)
+        {
+            for (uint32_t j = 0; j < _period; j += 2)
+                {
+                    _dest[i * _period + j] = _prn[i];
+                }
+            for (uint32_t j = 1; j < _period; j += 2)
+                {
+                    _dest[i * _period + j] = -_prn[i];
+                }
+        }
+}
+
+
+//! Generates float version of sine BOC(1,1) modulated Data Code 
+void beidou_b1cd_gen_float_11(gsl::span<float> _dest, gsl::span<int> _prn)
+{
+	const uint32_t _codeLength = 12 * BEIDOU_B1Cd_CODE_LENGTH_CHIPS;
```

Comment:
Use a variable for this or add a comment to this value. Reading 12 is not clear where you got that